app/signup.tsx

```diff
@@ -55,8 +60,14 @@ export default function SignupScreen() {
     watchedValues.confirmPassword
 
   const onSubmit = (data: SignupFormData) => {
-    console.log('Signup data:', data)
-    // TODO: Implement signup functionality
+    const signupData = {
+      email: data.email,
+      password: data.password,
+      confirmPassword: data.confirmPassword,
+      selectedGenres: selectedGenres,
```

Comment:
Is the sign up endpoint able to receive the selected genres? If not, you will need to update you BE